qiskit_aer/noise/noise_model.py

```diff
@@ -953,108 +951,6 @@ def to_dict(self, serializable=False):
-                    )
-                elif dic["name"] == "unitary":
-                    circ.append(UnitaryGate(data=dic["params"][0]), qargs=dic["qubits"])
-                elif dic["name"] == "pauli":
-                    circ.append(PauliGate(dic["params"][0]), qargs=dic["qubits"])
-                else:
-                    with catch_warnings():
-                        filterwarnings(
-                            "ignore",
-                            category=DeprecationWarning,
-                            module="qiskit_aer.noise.errors.errorutils",
-                        )
-                        circ.append(
-                            UnitaryGate(
-                                label=dic["name"], data=_standard_gate_unitary(dic["name"])
-                            ),
-                            qargs=dic["qubits"],
-                        )
-            return circ
-
-        # Return noise model
-        noise_model = NoiseModel()
-
-        # Get error terms
-        errors = noise_dict.get("errors", [])
-
-        for error in errors:
-            error_type = error["type"]
-
-            # Add QuantumError
-            if error_type == "qerror":
-                circuits = [inst_dic_list_to_circuit(dics) for dics in error["instructions"]]
-                noise_ops = tuple(zip(circuits, error["probabilities"]))
-                qerror = QuantumError(noise_ops)
-                qerror._id = error.get("id", None) or qerror.id
-                instruction_names = error["operations"]
-                all_gate_qubits = error.get("gate_qubits", None)
-                if all_gate_qubits is not None:
-                    for gate_qubits in all_gate_qubits:
-                        # Add local quantum error
-                        noise_model.add_quantum_error(
-                            qerror, instruction_names, gate_qubits, warnings=False
-                        )
-                else:
-                    # Add all-qubit quantum error
-                    noise_model.add_all_qubit_quantum_error(
-                        qerror, instruction_names, warnings=False
-                    )
-
-            # Add ReadoutError
-            elif error_type == "roerror":
-                probabilities = error["probabilities"]
-                all_gate_qubits = error.get("gate_qubits", None)
-                roerror = ReadoutError(probabilities)
-                # Add local readout error
-                if all_gate_qubits is not None:
-                    for gate_qubits in all_gate_qubits:
-                        noise_model.add_readout_error(roerror, gate_qubits, warnings=False)
-                # Add all-qubit readout error
-                else:
-                    noise_model.add_all_qubit_readout_error(roerror, warnings=False)
-            # Invalid error type
-            else:
-                raise NoiseError("Invalid error type: {}".format(error_type))
-        return noise_model
```

Comment:
Is it difficult to implement `from_dict` for new `BaseQuantumError` structure?
(I think `from_dict` is not so important, but I wonder if there are some users need this)

Reply:
I see

Reply:
For what it's worth, this is useful for people who want to run noisy simulations on a server but want to enable clients to pick their noise model. I'll also note that the associated `to_dict` method is still present.

I'm not using this currently and I'm not requesting it be restored. The reason I noticed is because a unit test of mine broke with qiskit-aer 0.15, but I'll just delete my test.

Reply:
This removal also broke the unit tests in `qiskit-ibm-runtime` because we need to serialize/deserialize noise models to send to the server.

For now, we can pin the aer version while we deprecate and remove support for this on our side.